src/spdx2/agent/spdx2utils.php

```diff
@@ -63,7 +63,11 @@ static public function addPrefixOnDemand($license, $spdxValidityChecker = null)
       }
       else
       {
-        return self::$prefix . $license;
+        if(!empty($license)){
+          return self::$prefix . $license;
+        }else{
+          return "NOASSERTION";
```

Comment:
I think it would be more consistent to call the function with `addPrefixOnDemand("NOASSERTION")` instead of an empty string and add
```
    if($license === "NOASSERTION" || $license === "NONE"){
        return $license;
    }
```
to the top of `addPrefixOnDemand`.